code/modules/mob/living/carbon/carbon.dm

```diff
@@ -259,14 +259,15 @@
 			else
 				var/mob/living/carbon/human/hugger = M
 				if(M.resting == 1) //Are they resting on the ground?
-					M.visible_message("<span class='notice'>[M] grabs onto [src] and pulls [M.get_visible_gender() == MALE ? "himself" : get_visible_gender() == FEMALE ? "herself" : "themselves"] up off the ground!</span>", \
-							"<span class='notice'>You grip onto [src] and pull yourself up off the ground!</span>")
-					M.resting = 0 //Hoist yourself up up off the ground. No para/stunned/weakened removal.
+					M.visible_message("<span class='notice'>[M] grabs onto [src] and pulls \himself </span>", \
+							"<span class='notice'>You grip onto [src] and pull yourself up off the ground!</span>") //AHHH gender checks are hard, but this should work	
 					if(M.fire_stacks >= (src.fire_stacks + 3)) //Fire checks.
 						src.adjust_fire_stacks(1)
 						M.adjust_fire_stacks(-1)
 					if(M.on_fire)
 						src.IgniteMob()
+					sleep(5) //.5 second delay. Makes it a bit stronger than just typing rest.

```

Comment:
Genuinely unsure, but does a do_after not work here?

Reply:
I'm not sure. I just used sleep as I've seen it used many times before and I know it'd work here. I can see if do_after works.

Reply:
It may not work, I don't know if do_after() works for resting mobs.